src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -905,6 +908,12 @@ public void run() {
                         + "I won't be able to participate in leader "
                         + "election any longer: "
                         + self.getElectionAddress());
+                if (exitException instanceof BindException) {
+                    // After leaving listener thread, the host cannot join the
+                    // quorum anymore, this is a severe error that we cannot
+                    // recover from, so we need to exit
+                    System.exit(14);
```

Comment:
how did we pick 14?

Reply:
It would be great to track these exit codes at a common place somewhere. Would you mind creating constant fields for them?
If there're too many, we might want to create a separate Jira for the refactoring.
Documenting them would also be very helpful. It seems like we don't list them in the docs yet, so that might also be part of the new Jira.